src/clj/harja/palvelin/komponentit/liitteet.clj

```diff
@@ -98,61 +219,102 @@
       (io/copy alkuperainen temp-file)
       [(io/input-stream temp-file) (io/input-stream temp-file)])))
 
-(defn- tallenna-liite [db fileyard-client tiedostopesula virustarkistus luoja urakka tiedostonimi tyyppi uskoteltu-koko lahdetiedosto kuvaus lahde-jarjestelma]
+(defn- tallenna-liite-tietokantaan [db lahdetiedosto alusta s3-url fileyard-client liite-perustiedot]
```

Comment:
Funktiossa paljon tykitystä tuohon perustiedot mappiin. Jos käytettäisiin destrukturointia? 
Olisiko tuohon (not (nil?)) mielummin (some?)? 

Halutaanko tuossa siis katsoa että joko oid, s3hash tai fileyard-hash olemassa? Voisi lyödä myös pari line breakkia ettei näytä kauhean mössöltä, ainakaan sen enempää mitä clojure jo on 😂

Eli jotenkin näin, ehkä:
```
(defn- tallenna-liite-tietokantaan [db lahdetiedosto alusta s3-url fileyard-client 
                                    {:keys [tiedostonimi tyyppi koko luoja urakka kuvaus lahdejarjestelma] :as liite-perustiedot}]
  (let [pikkukuva (muodosta-pikkukuva (io/input-stream lahdetiedosto))
        data (tallenna-liitteen-data db alusta s3-url fileyard-client lahdetiedosto tiedostonimi)
        liite (when (or 
                      (some? (:liite_oid data))
                      (some? (:s3hash data))
                      (some? (:fileyard-hash data)))
                (liitteet-q/tallenna-liite<!
                  db
                  (merge {:nimi tiedostonimi
                          :tyyppi tyyppi
                          :koko koko
                          :pikkukuva pikkukuva
                          :luoja luoja
                          :urakka urakka
                          :kuvaus kuvaus
                          :lahdejarjestelma lahdejarjestelma}
                    data)))]
    (if liite
      (do
        (log/info "Liite tallennettu.")
        liite)
      (throw+ {:type virheet/+ominaisuus-ei-kaytossa+ :virheet
               [{:koodi virheet/+ominaisuus-ei-kaytossa+
                 :viesti (str "Liitteen tallennus ei tällä hetkellä onnistu. Kokeile myöhemmin uudestaan.")}]}))))
```

Reply:
Sillä tykityksellä just siivottiin ruuhkaa, joka tuli funktion parametreissa. :) 
Jossain vaiheessa ne voisi destructuroida ja harkitsinkin sitä, mutta päätin tehdä sen lopulta noin yksitellen.

Jos somea käyttäisi niin näkisin, että se pitäisi tehdä jotenkin tyyliin (every? some? (ja tähän vektorina ne arvot)) 
mutta tuota tehdessä päädyin siihen, että nil tarkistus saa nyt riittää :)

Reply:
Otin vielä nuo some? tarkistukset matkaan